src/SettingsProvider.js

```diff
@@ -1,7 +1,13 @@
 import PropTypes from 'prop-types';
 import React, { createContext, useState } from 'react';
 
-export const SettingsContext = createContext();
+export const SettingsContext = createContext({
+  globalSettings: {
+    filter: {},
+    sections: {},
+    widgets: []
+  }
+});
```

Comment:
good catch from tests, that these were undefined initially...i think this cannot break any logics for the app. please have a short thought about it @digorath

Reply:
widgets doesn't need to be defined, the component handles the undefined case itself.

as far as i could see the sections always have `{}` as its default value, wherever they are used. same with filter.

But why is `navigation` set as an empty object in the new versions? it should be either `consts.DRAWER` or `consts.TABS`

Reply:
what does the test run say when removing widgets here?
sections was necessary, otherwise it failed.
good point to set navigation to drawer. but an empty initial value would result in that as well through code, right?

```
if (_isEmpty(globalSettings)) {
  globalSettings = {
    navigation: consts.DRAWER
  };
}
```

Reply:
```
globalSettings: {
  filter: {},
  navigation: consts.DRAWER,
  sections: {}
}
```

makes the tests also pass. take this?

Reply:
👍

Reply:
122ee8e